src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl.xaml.cs

```diff
@@ -0,0 +1,229 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Windows;
+using System.Windows.Controls;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.LanguageServices.Implementation;
+using Microsoft.VisualStudio.LanguageServices.Implementation.LanguageServiceBrokerShim;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    /// <summary>
+    /// Interaction logic for DocumentOutlineControl.xaml
+    /// </summary>
+    internal partial class DocumentOutlineControl : UserControl, IVsCodeWindowEvents
+    {
+        private ILanguageServiceBrokerShim LanguageServiceBroker { get; }
+
+        private IThreadingContext ThreadingContext { get; }
+
+        private IVsEditorAdaptersFactoryService EditorAdaptersFactoryService { get; }
+
+        private IVsCodeWindow CodeWindow { get; }
+
+        /// <summary>
+        /// The type of sorting to be applied to the UI model in <see cref="UpdateUIAsync"/>.
+        /// </summary>
+        private SortOption SortOption { get; set; }
```

Comment:
nit, for mutable data, it can be good to mention which threads it is safe to mutate/read this from.  my guess is that both are only safe from the UI thread, but i don't know.  Also, if it is only safe on the UI thread, you could actually expand the property to assert that to catch issues.